server/src/main/java/org/opensearch/snapshots/RestoreService.java

```diff
@@ -752,12 +830,17 @@ private IndexMetadata updateIndexSettings(
                         Settings.Builder settingsBuilder = Settings.builder()
                             .put(settings.filter(settingsFilter))
                             .put(normalizedChangeSettings.filter(k -> {
-                                if (UNMODIFIABLE_SETTINGS.contains(k)) {
+                                if (USER_UNMODIFIABLE_SETTINGS.contains(k)) {
                                     throw new SnapshotRestoreException(snapshot, "cannot modify setting [" + k + "] on restore");
                                 } else {
                                     return true;
                                 }
                             }));
+
+                        // override internal settings
+                        if (changeSettingsInternal != null) {
+                            settingsBuilder.put(changeSettingsInternal).normalizePrefix(IndexMetadata.INDEX_SETTING_PREFIX);
+                        }
```

Comment:
It appears to me that the settings in the `changeSettingsInternal` have the index prefix present always. Why do we need to normalise prefix again?